pkg/bootstrapper/cloud/aws.go

```diff
@@ -0,0 +1,35 @@
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+)
+
+// getAWSMetadataHostname returns name of the AWS host from metadata service
+func getAWSMetadataHostname() (string, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return "", fmt.Errorf("unable to load config: %w", err)
+	}
+
+	client := imds.NewFromConfig(cfg)
+
+	hostnameRes, err := client.GetMetadata(context.TODO(), &imds.GetMetadataInput{
+		Path: "hostname",
+	})
```

Comment:
Is there a reference somewhere for why we are using this value? Perhaps a link to the old cloud provider code or something? It would be good to link the original source if possible